src/flagsmith/mod.rs

```diff
@@ -280,15 +275,23 @@ impl Flagsmith {
         return Ok(flags);
     }
 
-    fn build_identity_model(
+    fn get_identity_model(
         &self,
         environment: &Environment,
+        identities_with_overrides_by_identifier: &HashMap<String, Identity>,
         identifier: &str,
         traits: Vec<Trait>,
     ) -> Result<Identity, error::Error> {
-        let mut identity = Identity::new(identifier.to_string(), environment.api_key.clone());
+        let mut identity: Identity;
+
+        if identities_with_overrides_by_identifier.contains_key(identifier) {
+            identity = identities_with_overrides_by_identifier.get(identifier).unwrap().clone();
+        } else {
+            identity = Identity::new(identifier.to_string(), environment.api_key.clone());
+        }
```

Comment:
In both cases I can see that we use this method, we already have the identifier and it seems odd to me to pass the whole `identities_with_overrides_by_identifier` in along with the identifier. Couldn't we factor this out somehow?

Reply:
I sorta disagree. Some chunk of logic is going to have to retrieve the identity and I would rather have it in a specialized function.

Reply:
> I would rather have it in a specialized function.

Yep, I applied the same logic here. I do have to pass the `identities_with_overrides_by_identifier` around because of the top-level datastore locking routine, which happens in multiple places:

https://github.com/Flagsmith/flagsmith-rust-client/blob/c2ab95ce51e97d71d51ca5a1df9887a0c02d8d4d/src/flagsmith/mod.rs#L164-L165
https://github.com/Flagsmith/flagsmith-rust-client/blob/c2ab95ce51e97d71d51ca5a1df9887a0c02d8d4d/src/flagsmith/mod.rs#L193-L198
https://github.com/Flagsmith/flagsmith-rust-client/blob/c2ab95ce51e97d71d51ca5a1df9887a0c02d8d4d/src/flagsmith/mod.rs#L207-L212

Reply:
Ok, that's fine, I don't feel strongly enough here either way so I'll resolve this conversation.